### PR TITLE
docs: Update custom pod specs page

### PR DESCRIPTION
**Pull request by tara-det-ai**
Update custom pod specs page to include the new capability (0.23.4):

Kubernetes: Users may now populate all `securityContext` fields within the pod spec of the `determined-container` container except for `RunAsUser` and `RunAsGroup`. For those fields, use `det user link-with-agent-user` instead.

### CODE REVIEW

File: docs/setup-cluster/deploy-cluster/k8s/custom-pod-specs.rst [hunk 1]
```diff
@@ -60,14 +63,37 @@ Determined does not currently support configuring:
...
+
+Determined supports configuring the following fields:
 
 -  Resource requests and limits (except GPU resources).
+
 -  Volume mounts and volumes.
 
+-  All ``securityContext`` fields within the pod spec of the ``determined-container`` container
+   except for ``RunAsUser`` and ``RunAsGroup``. For those fields, use ``det user
+   link-with-agent-user`` instead.
```
Comment (jgongd) [inline docs/setup-cluster/deploy-cluster/k8s/custom-pod-specs.rst:81]: Nit: do you mean to add a line break  user and  link-with-agent-user?
image

Comment (tara-det-ai) [reply docs/setup-cluster/deploy-cluster/k8s/custom-pod-specs.rst:81]: fixed

Comment (tara-det-ai) [reply docs/setup-cluster/deploy-cluster/k8s/custom-pod-specs.rst]: really fixed this time :-;

**Review by jgongd — APPROVED** [review]
LGTM

**Review by rb-determined-ai — APPROVED** [review]

**Pull request merged**